.github/workflows/ci.yml

```diff
@@ -10,57 +10,138 @@ on:
+env:
+  IMAGE_NAME: openformulieren/open-forms-ext-stuf-zds-payments
+
 jobs:
   tests:
+    name: Run the Django test suite
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python: ['3.10', '3.11', '3.12']
-        django: ['4.2']
-
-    name: Run the test suite (Python ${{ matrix.python }}, Django ${{ matrix.django }})
+    services:
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        ports:
+          - 5432:5432
+        # Needed because the postgres container does not provide a healthcheck
+        options:
+          --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+          --name postgres
+      redis:
+        image: redis:6
+        ports:
+          - 6379:6379
 
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+      - name: Checkout Open Forms
+        uses: actions/checkout@v3
+        with:
+          repository: open-formulieren/open-forms
+          path: open-forms
+
+      - name: Checkout StUF-ZDS payments extension
+        uses: actions/checkout@v3
+        with:
+          path: extension
+
+      - name: Set up backend environment
+        uses: maykinmedia/setup-django-backend@v1.1
         with:
-          python-version: ${{ matrix.python }}
+          apt-packages: 'libxml2-dev libxmlsec1-dev libxmlsec1-openssl gettext postgresql-client gdal-bin'
+          python-version: '3.12'
+          optimize-postgres: 'yes'
+          pg-service: 'postgres'
+          setup-node: 'yes'
+          nvmrc-custom-dir: 'open-forms'
+          npm-ci-flags: '--legacy-peer-deps'
+          working-directory: ${{ github.workspace }}/open-forms
 
-      - name: Install dependencies
-        run: pip install tox tox-gh-actions
+      - name: Make symlink in OF to the extension
+        run: |
+          ln -s ${{ github.workspace }}/extension/stuf_zds_payments ${{ github.workspace }}/open-forms/src
 
       - name: Run tests
-        run: tox
+        run: |
+          export OPEN_FORMS_EXTENSIONS=stuf_zds_payments
+
+          python src/manage.py compilemessages
+          coverage run --source=stuf_zds_payments src/manage.py test stuf_zds_payments
+          coverage combine
+          coverage xml -o coverage-extension.xml
         env:
-          PYTHON_VERSION: ${{ matrix.python }}
-          DJANGO: ${{ matrix.django }}
+          DJANGO_SETTINGS_MODULE: openforms.conf.ci
+          SECRET_KEY: dummy
+          DB_USER: postgres
+          DB_PASSWORD: ''
+        working-directory: ${{ github.workspace }}/open-forms
 
       - name: Publish coverage report
-        uses: codecov/codecov-action@v4
+        uses: codecov/codecov-action@v3.1.4
         with:
-          token: ${{ secrets.CODECOV_TOKEN }}
+          root_dir: ${{ github.workspace }}/extension
+          working-directory: ${{ github.workspace }}/open-forms
+          files: ./coverage-extension.xml
+
+  docker_build:
```

Comment:
Looking at this, I'm wondering a little bit how useful it is to do this docker build in the first place - it doesn't make sense to publish this image by itself, it makes more sense to install the extra libraries with extensions instead, like we do in an OF build.

So feel free to remove this step - I think it used to be relevant when we did the (unit) testing in Docker, but the checkout + symlink approach replaced that.